docs/understanding-airbyte/database-data-catalog.md

```diff
@@ -0,0 +1,72 @@
+* `actor_definition`
+  * Each record represents a connector that Airbyte supports, e.g. Postgres. This table represents all the connectors that is supported by the current running platform.
+  * The `actor_type` column tells us whether the record represents a Source or a Destination.
+  * The `spec` column is a JSON blob. The schema of this JSON blob matches the [spec](airbyte-protocol.md#actor-specification) model in the Airbyte Protocol. Because the protocol object is JSON, this has to be a JSON blob.
+  * The `release_stage` describes the certification level of the connector (e.g. Alpha, Beta, Generally Available).
+  * The `docker_repository` field is the name of the docker image associated with the connector definition. `docker_image_tag` is the tag of the docker image and the version of the connector definition.
+  * The `source_type` field is only used for Sources, and represents the category of the connector definition (e.g. API, Database).
+  * The `resource_requirements` field sets a default resource requirement for any connector of this type. This overrides the default we set for all connector definitions, and it can be overridden by a connection-specific resource requirement. The column is a JSON blob with the schema defined in [ActorDefinitionResourceRequirements.yaml](airbyte-config/config-models/src/main/resources/types/ActorDefinitionResourceRequirements.yaml)
+  * The `public` boolean column, describes if a connector is available to all workspaces or not. For non, `public` connector definitions, they can be provisioned to a workspace using the `actor_definition_workspace_grant` table. `custom` means that the connector is written by a user of the platform (and not packaged into the Airbyte product).
+  * Each record contains additional metadata and display data about a connector (e.g. `name` and `icon`), and we should add additional metadata here over time.
+* `actor_definition_workspace_grant`
+  * Each record represents provisioning a non `public` connector definition to a workspace.
+  * todo (cgardens) - should this table have a `created_at` column?
+* `actor`
+  * Each record represents a configured connector. e.g. A Postgres connector configured to pull data from my database.
+  * The `actor_type` column tells us whether the record represents a Source or a Destination.
+  * The `actor_definition_id` column is a foreign key to the connector definition that this record is implementing.
+  * The `configuration` column is a JSON blob. The schema of this JSON blob matches the schema specified in the `spec` column in the `connectionSpecification` field of the JSON blob. Keep in mind this schema is specific to each connector (e.g. the schema of Postgres and Salesforce are different), which is why this column has to be a JSON blob.
+* `actor_catalog`
+  * Each record contains a catalog for an actor. The records in this table are meant to be immutable.
```

Comment:
does this mean a connector catalog - as in, the streams/fields that are defined for a particular connector? Are they actually immutable? What if streams are added/removed/modified?

Reply:
we can always fetch a new catalog for a particular connector. when we do, we will then use the new catalog instead of the old one. this is saying that we will keep around the old catalogs. we never mutate the old ones.